common/common.go

```diff
@@ -436,27 +508,34 @@ func RunBuild(verbose bool, dir, imageName, dockerfile string, buildArgs []strin
 	}
 
 	go func(done chan<- error) {
-		args := []string{
-			"build",
-			"-t", imageName,
-			"-f", dockerfile,
-		}
-		if noCache {
-			args = append(args, "--no-cache")
-		}
+		var dockerBuildCmdArgs []string
+		// Depending whether architecture list is passed or not trigger docker buildx or docker build accordingly
+		if architectures != nil && len(architectures) != 0 {
+			err := initializeContainerBuilder(containerEngineType)
+			if err != nil {
+				done <- err
+				return
+			}
 
-		if len(buildArgs) > 0 {
-			for _, buildArg := range buildArgs {
-				args = append(args, "--build-arg", buildArg)
+			var mappedArchitectures []string
+			for _, arch := range architectures {
+				mappedArch, ok := architectureMap[arch]
+				if !ok {
+					err = errors.New(fmt.Sprintf("invalid architecture type : %v", arch))
+					done <- err
+					return
+				}
+				mappedArchitectures = append(mappedArchitectures, mappedArch)
 			}
+
+			dockerBuildCmdArgs = buildXDockerCommand(imageName, dockerfile,  buildArgs, noCache, mappedArchitectures)
+			// perform cleanup
+			defer cleanupContainerBuilder(containerEngineType)
+		} else {
+			dockerBuildCmdArgs = buildDockerCommand(imageName, dockerfile,  buildArgs, noCache)
 		}
-		args = append(args,
-			"--build-arg", "HTTP_PROXY",
-			"--build-arg", "HTTPS_PROXY",
-			".")
 
-		// Container engine type would be optional here
-		cmd := exec.Command(containerEngineType, args...)
+		cmd := exec.Command("docker", dockerBuildCmdArgs...)
```

Comment:
Can we restore it to containerEngineType? It helps to switch across docker or podman in future

Reply:
Done.